exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -41,71 +36,64 @@ func (ke kafkaErrors) Error() string {
 	return fmt.Sprintf("Failed to deliver %d messages due to %s", ke.count, ke.err)
 }
 
-func (e *kafkaTracesProducer) tracesPusher(ctx context.Context, td ptrace.Traces) error {
-	messages, err := e.marshaler.Marshal(td, getTopic(ctx, &e.cfg, td.ResourceSpans()))
+type kafkaMessager[T any] interface {
+	partitionData(T) iter.Seq2[[]byte, T]
```

Comment:
Do you mind expanding why it returns `T`?

Reply:
Oh, it returns the key.

Reply:
I'll add a doc comment